test/unit/process/test_process.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2023 National Center for Atmospheric Research,
+// SPDX-License-Identifier: Apache-2.0
+#include <gtest/gtest.h>
+
+#include <micm/process/arrhenius_rate_constant.hpp>
+#include <micm/process/process.hpp>
+#include <micm/process/surface_rate_constant.hpp>
+#include <micm/process/user_defined_rate_constant.hpp>
+#include <micm/util/matrix.hpp>
+#include <micm/util/vector_matrix.hpp>
+#include <random>
+
+template<template<class> class MatrixPolicy>
+void testProcessUpdateState(const std::size_t number_of_grid_cells)
+{
+  micm::Species foo("foo", { { "molecular weight [kg mol-1]", 0.025 }, { "diffusion coefficient [m2 s-1]", 2.3e2 } });
+  micm::ArrheniusRateConstant rc1({ .A_ = 12.2, .C_ = 300.0 });
+  micm::SurfaceRateConstant rc2({ .label_ = "foo_surf", .species_ = foo });
+  micm::UserDefinedRateConstant rc3({ .label_ = "bar_user" });
+
+  micm::Process r1 = micm::Process::create().rate_constant(rc1);
+  micm::Process r2 = micm::Process::create().rate_constant(rc2);
+  micm::Process r3 = micm::Process::create().rate_constant(rc3);
+  std::vector<micm::Process> processes = { r1, r2, r3 };
+
+  std::vector<std::string> param_labels{};
+  for (const auto& process : processes)
+    for (auto& label : process.rate_constant_->CustomParameters())
+      param_labels.push_back(label);
+  micm::State<MatrixPolicy> state{ micm::StateParameters{ .custom_rate_parameter_labels_ = param_labels,
+                                                          .number_of_grid_cells_ = number_of_grid_cells,
+                                                          .number_of_rate_constants_ = processes.size() } };
+
+  MatrixPolicy<double> expected_rate_constants(number_of_grid_cells, 3, 0.0);
+  std::vector<double> params = { 0.0, 0.0, 0.0 };
+  auto get_double = std::bind(std::lognormal_distribution(0.0, 0.01), std::default_random_engine());
+
+  for (std::size_t i_cell = 0; i_cell < number_of_grid_cells; ++i_cell)
+  {
+    state.conditions_[i_cell].temperature_ = get_double() * 285.0;
+    state.conditions_[i_cell].pressure_ = get_double() * 101100.0;
+    params[0] = get_double() * 1.0e-8;
+    params[1] = get_double() * 1.0e5;
+    params[2] = get_double() * 1.0e-2;
```

Comment:
At some point, I'd like to see if we could come up with a different API for this. Perhaps we could do something that sets these parameters similar to how the user defined rate constants are set on the state.

Reply:
That's what these parameters are - I was just saving them in a vector so I could directly calculate the expected results. But I think they are being used as you describe here: https://github.com/NCAR/micm/blob/b5a2c6bb9550c01c9fe0d0a1b41d1b72a367731b/test/unit/process/test_process.cpp#L45-L49